### DISCUSSION TITLE
Consider GPG signing the release tags

### DISCUSSION THREAD

**Discussion by inglor** [polybar/polybar#2718] [Ideas]
Hi,

I'm the packager for Arch Linux `polybar` on the official repos. Please consider gpg signing the tags of the releases so it can be used while building the package. This will keep the chain of trust between the commits / tags and package available on repos.

Cheers,
Leonidas

**Comment by patrick96** [comment]
Just got the email that my AUR package was deleted ;)

Do you need the git tag signed or the release tarball (or both)?

**Reply by inglor** [reply]
Git tag signed would be enough. Also if you could mentioned your GPG public key on the readme that would be handy as well. Signing the tarball is not required if the tag is signed because I will be using something like the below and `makepkg` knows to fetch the git repo and verify the tag with the provided PGP public key: 
```
source=("git+https://github.com/polybar/polybar.git#tag=v${pkgver}?signed")
```
You can check an example like the below:
https://github.com/archlinux/svntogit-community/blob/packages/memray/trunk/PKGBUILD

**Reply by patrick96** [reply]
Alright, I'll update our release workflow to produce signed git tags. Release should then be signed starting with the next release.

Also, be aware that we have two git submodules in the repo (`lib/i3ipcpp` and `lib/xpp`).

**Reply by inglor** [reply]
Ah fun - submodules are always fun (not) when using the repo :) Maybe signing the tarball is not a bad idea after all (if it contains the submodules) :stuck_out_tongue:

**Reply by patrick96** [reply]
That's precisely why we provide the tarball ;)

I'll upload a signature for the 3.6.3 release as well.

**Reply by patrick96** [reply]
Signature is uploaded. I signed using my gpg key `1D5791352D51A228D4DDDBA4521E5E03AEBCA1A7`. I'll add proper documentation later.

**Reply by patrick96** [reply]
Proper documentation is now in place as well: https://github.com/polybar/polybar#signatures
